runtime-transaction/src/runtime_transaction.rs

```diff
@@ -85,12 +85,14 @@ impl RuntimeTransaction<SanitizedVersionedTransaction> {
         let is_simple_vote_tx = is_simple_vote_tx
             .unwrap_or_else(|| is_simple_vote_transaction(&sanitized_versioned_tx));
 
-        let precompile_signature_details = get_precompile_signature_details(
-            sanitized_versioned_tx
-                .get_message()
-                .program_instructions_iter()
-                .map(|(program_id, ix)| (program_id, SVMInstruction::from(ix))),
-        );
+        let (precompile_signature_details, compute_budget_instruction_details) =
```

Comment:
@tao-stones this is top-level processing function. Single loop over the instructions that we can use to get all the details we want. **very** easy to add more to this!